app/services/recent-events.ts

```diff
@@ -164,10 +166,20 @@ export class RecentEventsService extends StatefulService<IRecentEventsState> {
 
   syncEventsState() {
     this.formEventsArray();
-    this.websocketService.socketEvent.subscribe(this.onSocketEvent.bind(this));
+    this.makeSocketConnection();
     return this.fetchMutedState();
   }
 
+  makeSocketConnection() {
+    if (this.socketConnection) {
+      this.socketConnection.unsubscribe();
+    }
+
+    this.socketConnection = this.websocketService.socketEvent.subscribe(
+      this.onSocketEvent.bind(this),
```

Comment:
Don't use `.bind` since it loses typechecking. Use arrow functions instead. 
```
this.socketConnection = this.websocketService.socketEvent.subscribe(ev => this.onSocketEvent(ev))
```